Warm Bubble

OsKnoth opened the pull request:
Implemented warm bubble, added two new fucntions to compressible_euler.jl

and a new toml

OsKnoth committed:
Warm Bubble

Implemented warm bubble, added two new fucntions to compressible_euler.jl

and a new toml

OsKnoth commented:
![visit0000](https://user-images.githubusercontent.com/50015520/92476147-cc3ffa00-f1de-11ea-863c-2d4747b2546b.png)
![visit0001](https://user-images.githubusercontent.com/50015520/92476154-cd712700-f1de-11ea-9111-cc20b24199da.png)
![visit0002](https://user-images.githubusercontent.com/50015520/92476158-ce09bd80-f1de-11ea-9a1b-b7acea3eee53.png)
![visit0003](https://user-images.githubusercontent.com/50015520/92476160-cea25400-f1de-11ea-9215-78869b973c22.png)

OsKnoth commented:
ASAM simulations with a finite volume code after 500 s and 1000 s (final time), vertical velocity and potential temperature

OsKnoth commented:
![visit0005](https://user-images.githubusercontent.com/50015520/92476415-335dae80-f1df-11ea-9899-4d9706dd3eed.png)
![visit0006](https://user-images.githubusercontent.com/50015520/92476419-348edb80-f1df-11ea-9336-fccf837a7c2e.png)

OsKnoth commented:
Trixi simulations. vertical velocity,

OsKnoth committed:
Pot Temp

New function cons2pot

cons2prim replaced by cons2pot

OsKnoth commented:
![visit0009](https://user-images.githubusercontent.com/50015520/92482280-bbe04d00-f1e7-11ea-9f7d-b7c3c8935e75.png)
![visit0008](https://user-images.githubusercontent.com/50015520/92482304-c4388800-f1e7-11ea-8bce-9f84bb0d4a69.png)
![visit0010](https://user-images.githubusercontent.com/50015520/92482329-cd295980-f1e7-11ea-8c0d-1af0efecfa5e.png)

OsKnoth committed:
Inline function for potential temperature

Inline function for potential temperature in compressible_euler.jl

OsKnoth committed:
New limiter function lmars

Simple limiter for low Mach number flows

gregorgassner commented:
It would be maybe a good idea (for future) to make a new equation system out of it...maybe compressible Euler equations directly formulated in density, momentum and potential temperature instead of density, momentum, total energy?

andrewwinters5000 commented:
> It would be maybe a good idea (for future) to make a new equation system out of it...maybe compressible Euler equations directly formulated in density, momentum and potential temperature instead of density, momentum, total energy?

I agree. I think that is why the numerical solution has so much noise. I was reading through the reference for the warm bubble testcase and the fluxes in the potential temperature are different than those for the total energy. I also believe that this causes a bug in `flux_larms` because the current implementation still has the total energy as the fourth variable in `u_ll` and `u_rr` but the numerical flux is built assuming the fourth variable is potential temperature (see Eqs. 46 and 47 in the reference from the docstring)

OsKnoth committed:
lmars corrected

there was an error in the energy,
but again there are problems with the lower and upper boudary conditions

OsKnoth commented:
Yes there was an error in the energy which I corrected, but there a further problems with lower and upper boundary conditions

sloede commented:
I have an idea for a way to bring in point-wise state-aware Dirichlet BCs that would be comparatively simple to implement. It wouldn't work for BC types where information about the entire boundary element or even the complete side is required. But if y'all think that point-wise would be good enough (at least for the time being), I can think about whether this can be done quickly next week.

sloede commented:
@OsKnoth I like the idea by @gregorgassner to add a new system of equations that uses the potential temperature as a conservative variable. This could then also be a good starting point for adding things like "wet air" etc.

OsKnoth committed:
Bubble works

flux function lmars changed with respect to energy
new boundary conditions, no flux condition
new output option

OsKnoth commented:
New results for the bubble
![visit0011](https://user-images.githubusercontent.com/50015520/92699936-7c804080-f34e-11ea-84ca-696004d6c0ec.png)

OsKnoth committed:
Euler equation with potential temperature instead of energy

new equation file
new toml file
works for weak formulation and lmars boundary fluxes

sloede commented:
@OsKnoth just a quick update: In #186 we are close to converging on an approach that would allow to properly specify boundary conditions. Once this is merged to `master` (hopefully by the beginning of next week), we should merge `master` to your branch and update this PR with proper slip-wall boundary conditions.

sloede commented:
@OsKnoth #186 is now merged, thus it should be possible to implement the boundary conditions for the upper and lower boundary in a proper manner. Please let me know if I can assist with merging `master` to your branch, as there will be merge conflicts. I'd be happy to help, though, either via Zoom or by doing the merge myself (for the latter you'd have to add me as a collaborator to your fork such that I can make changes there directly).

OsKnoth committed:
Moist Pot Temp

Included density virtual potential temperature,
Initial Profile for the Bryan-Fritsch test case

OsKnoth committed:
Intermediate Error seeking

Moist bubble is working, but with the false height

OsKnoth committed:
Moist bubble works

Dry and moist bubble in one configuration

OsKnoth committed:
Moist bubble with energy formualtion

There i snow a version for a moist bubble with enrgy formulation

sloede committed:
Remove Trixi2Img/Trixi2Vtk from dependencies

sloede committed:
Move using NLsolve to Trixi.jl

sloede committed:
Merge branch 'master' into warm_bubble

sloede committed:
Fix solution output for potential temperature variables

sloede committed:
Add boundary_conditions_slip_wall

sloede committed:
Remove obsolete Project.toml

OsKnoth committed:
New MoistPotTemp and MoistEnergy

Cleaned up the moist (including dry) test cases for energy and potenetial temperature

sloede committed:
Remove erroneously added file

sloede committed:
Remove erroneously added file

sloede committed:
Remove erroneously added file

sloede commented on src/equations/2d/compressible_euler_pot.jl:
```diff
@@ -0,0 +1,681 @@
+The compressible Euler equations for a  gas with dry air and water vapor and a condensed phase 
+in two space dimensions.
+"""
+struct CompressibleMoistPotTempEulerEquations2D <: AbstractCompressibleEulerEquations{2, 6}
```
If I recall our conversation correctly, @OsKnoth, this system of equations was supposed to be just the Euler equations with potential temperature and dry air. However, it seems like this is virtually identical to the moist equations. Or am I misunderstanding something here?

sloede commented on src/equations/2d/compressible_euler_pot.jl:
```diff
@@ -0,0 +1,681 @@
+@doc raw"""
+    CompressibleMoistPotTempEulerEquations2D
```
I think we should rename this to `CompressibleEulerMoistEquations2D` (to easily see that it's compressible Euler + some special feature). What do you think @trixi-framework/principal-developers?

sloede commented on src/equations/2d/compressible_euler_moist.jl:
```diff
@@ -0,0 +1,664 @@
+@doc raw"""
+    CompressibleMoistEnergyEulerEquations2D
```
Same argument as below: I think this should be named `CompressibleEulerMoistEnergyEquations2D` or something similar.

sloede commented on src/equations/2d/compressible_euler.jl:
```diff
@@ -5,13 +5,28 @@
 The compressible Euler equations for an ideal gas in two space dimensions.
 """
 struct CompressibleEulerEquations2D <: AbstractCompressibleEulerEquations{2, 4}
+  c_p::Float64
+  c_v::Float64
+  R_d::Float64
+  kappa::Float64
   gamma::Float64
+  _grav::Float64
+  p0::Float64
+  a::Float64
```
This comment still applies

sloede commented on src/io/io.jl:
```diff
@@ -127,10 +127,18 @@ function save_solution_file(dg::AbstractDg, mesh::TreeMesh, time, dt, timestep,
     if solution_variables == "conservative"
       data = dg.elements.u
       varnames = varnames_cons(equation)
+    elseif solution_variables == "pot"
```
I think `pot` is not descriptive enough. The indicator could be named `pottemp` though, and the related functions to `cons2pottemp` and `varnames_pottemp`

ranocha commented:
Memo: @OsKnoth should be added to `AUTHORS.md` before this is merged.
